### DISCUSSION TITLE
running javascript files from remote server

### DISCUSSION THREAD

**Discussion by hsd-dev** [zauberzeug/nicegui#2439] [Q&A] [answered]
### Question

I have a `flask` server running which servers `HTML` pages with scripts stored in the static folder. This is an example script included in the header
```html
<script src="{{url_for('static', filename='js/viewer.js')}}"></script>
```
and the `GET` requests are handled in the method
```py
@app.get("/view/<string:id>")
def view(id):
    ...
    return render_template("view.html", id=id)
```
If I type the address in my browser, I get the page rendered as expected
```
http://0.0.0.0:5500/view/1234
```
Now I want to embed this HTML in a `nicegui` application (reference: https://github.com/zauberzeug/nicegui/discussions/487), for which I used `requests` library
```py
from nicegui import app, ui
import requests


r = requests.get('http://0.0.0.0:5500/view/1234')
ui.add_body_html(r.text)
ui.run()
```
But I get an error 
```
http://127.0.0.1:8080/static/js/viewer.js not found
```
`http://127.0.0.1:8080` is the address on which the `nicegui` application running. Also if I print out the text of the response, this is what I get
```html
<script src="/static/js/viewer.js"></script>
```
How can I correctly locate these javascript files?

**Comment by falkoschindler** [comment, marked as answer]
Cross post: https://stackoverflow.com/questions/77881022/incorrect-use-of-url-for-in-flask

**Reply by hsd-dev** [reply]
sorry about that. I initially posted on SO then realized maybe this is a better place.

**Reply by falkoschindler** [reply]
No worries, I just wanted to indicate to other contributors that there's already an ongoing discussion with a potential answer. 🙂

**Reply by hsd-dev** [reply]
Do you agree with the solution or there is a "nicer" way?

**Reply by falkoschindler** [reply]
I don't immediately see a better way. Maybe the Flask template can generate relative URLs without a fixed IP. But if not, modifying the HTML using Beautiful Soup is a reasonable solution.
